src/ZoneServer/Skills/Handlers/Swordsman/Peltasta/ShieldLob.cs

```diff
@@ -0,0 +1,213 @@
+				Send.ZC_NORMAL.SetPadModel(shieldMonster, "warrior_f_", shield.Id);
+			}
+			Send.ZC_NORMAL.Skill_ItemRotate(shieldMonster, 90);
+			Send.ZC_NORMAL.SpinObject(shieldMonster);
+			Send.ZC_NORMAL.Skill_SetActorHeight(shieldMonster, skillHandle, 22);
+			Send.ZC_NORMAL.AttachEffect(shieldMonster, "I_light004_violet", 1.5f);
+
+			// Begin the shield throw, throwing shield outwards to the far point
+			var movement = shieldMonster.Components.Get<MovementComponent>();
+			movement.SetFixedMoveSpeed(150f);
+			movement.SetMoveSpeedType(MoveSpeedType.Run);
+			var moveTime = movement.MoveTo(originPos.GetRelative(caster.Direction, 100f));
+			Send.ZC_NORMAL.Skill_EffectMovement(caster, skillHandle, originPos.GetRelative(caster.Direction, 100f), 150);
+			
+			var cancelToken = new CancellationTokenSource();
+			checkShieldCollision(skill, caster, shieldMonster, cancelToken);
+			await Task.Delay(moveTime);
+
+			// Reached destination, pause briefly before returning
+			await Task.Delay(TimeSpan.FromMilliseconds(500));
+
+			// Return trip back to player.
+			movement.SetFixedMoveSpeed(200f);
+			cancelToken.Cancel();
+			cancelToken = new CancellationTokenSource();
+			checkShieldCollision(skill, caster, shieldMonster, cancelToken);
+
+			// We loop over this a few times so the shield homes in on your location even if you're moving
+			for (int i = 0; i < 5; i++) 
+			{ 
+				moveTime = movement.MoveTo(caster.Position);
+				Send.ZC_NORMAL.Skill_EffectMovement(caster, skillHandle, caster.Position, 200);
+				await Task.Delay(moveTime.Divide(3));
+			}
+
+			// Should be close enough to your position, return to the player
+			moveTime = movement.MoveTo(caster.Position);
+			Send.ZC_NORMAL.Skill_EffectMovement(caster, skillHandle, caster.Position, 200);
+			await Task.Delay(moveTime);
+
+			// Shield has returned, remove the shield and restore the shield to your hand
+			cancelToken.Cancel();
+			Send.ZC_NORMAL.SkillPad(caster, skill, "Peltasta_ShieldLob2", originPos, caster.Direction, 0, 145.8735f, skillHandle, 30, false);
+			caster.Map.RemoveMonster(shieldMonster);			
+		}
+
+
+
+		/// <summary>
+		/// Checks to see if the Shield collides with any enemies
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private async void checkShieldCollision(Skill skill, ICombatEntity caster, Mob shieldMonster, CancellationTokenSource cancelToken)
+		{
+			// We have to track sr manually since this skill can hit targets sequentially
+			List<ICombatEntity> hitTargets = new List<ICombatEntity>();
+			var sr = (int)skill.Properties.GetFloat(PropertyName.SkillSR);
+			if (ZoneServer.Instance.Conf.World.DisableSDR)
+				sr = int.MaxValue;
+
+			while (!cancelToken.IsCancellationRequested)
+			{
+				var splashArea = new Circle(shieldMonster.Position, 40);
+
+				var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
```

Comment:
Would this skill benefit from `Mobs` supporting trigger areas? It seems like that would be easier than running a custom collision check.

Reply:
I don't know, I like that this is somewhat similar to how most skills do their collission check.

Reply:
What do you mean by that? For clarification, trigger areas are essentially just shapes that move around the map with the entity, and an event is called when they're triggered, aka if an entity enters them. That way you could just let em rip and react once something happens.

Reply:
I mean the collision here is handled by defining a splashArea and checking for targets within that splasharea, just like most other skills.  I'm sure it could be done with events, I'm just not sure it would be simpler.

Reply:
> not sure it would be simpler

Wanna do a bet?^^

Reply:
Of course, if you'd like to recode parts of the skill you're welcome to, I just don't think the collision check is that complicated.